### PR TITLE
feat: add deleteAttributes to PersistenceAdapter interface and delete…

### LINKED ISSUE #436: Delete Persistent Attributes including account ID
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[ ] Performance issue
[ ] Feature request
[X] Documentation issue or request
[ ] Other... Please describe:
</code></pre>

## Expected Behavior
I am struggling to find how I can delete persistent Attributes **including** the key, which is the amazon account ID by default.

## Current Behavior
I am doing it the following way:

```
attributesManager.setPersistentAttributes();
await attributesManager.savePersistentAttributes();
```
But then, the ID remains as Key, there are just no entries for it. I want to delete everything for a given user.

### PR DESCRIPTION
**Pull request by tianrenz**
…PersistentAttributes to AttributesManager interface

## Description

- Adding optional method ``deleteAttributes`` to ``PersistenceAdapter`` interface
- Adding optional method ``deletePersistentAttributes`` to ``AttributesManager`` interface
- Adding default implementation to `DynamoDbPersistenceAdapter` and `S3PersistenceAdapter`
- Adding unit tests
- Fix some typo

## Motivation and Context

fixes #436

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project
- [x] My change requires a change to the documentation
- [x] I have updated the documentation accordingly
- [x] I have read the **README** document
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] My commit message follows [Conventional Commit Guideline](https://conventionalcommits.org/)

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

[issues]: https://github.com/alexa/alexa-skills-kit-sdk-for-nodejs/issues
[license]: http://aws.amazon.com/apache2.0/
[cla]: http://en.wikipedia.org/wiki/Contributor_License_Agreement

### CODE REVIEW

File: ask-sdk-core/tst/attributes/AttributesManagerFactory.spec.ts [hunk 1]
```diff
@@ -189,7 +228,7 @@ describe('AttributesManagerFactory', () => {
 
     });
 
-    it('should thrown an error when trying to save persistent attributes without persistence manager', async() => {
+    it('should thrown an error when trying to save persistent attributes without persistence adapter', async() => {
```
Comment (nikhilym) [inline ask-sdk-core/tst/attributes/AttributesManagerFactory.spec.ts:231]: nit: `throw`

**Review by nikhilym — APPROVED** [review]
LGTM :shipit:

File: ask-sdk-core/tst/attributes/AttributesManagerFactory.spec.ts (hunk 1 above)
Comment (tianrenz) [reply ask-sdk-core/tst/attributes/AttributesManagerFactory.spec.ts:231]: will fix in the next PR with some other minor fixes

File: ask-sdk-s3-persistence-adapter/lib/attributes/persistence/S3PersistenceAdapter.ts [hunk 2]
```diff
@@ -109,4 +109,22 @@ export class S3PersistenceAdapter implements PersistenceAdapter {
...
+            Bucket : this.bucketName,
+            Key : objectId,
+        };
+
+        try {
+            await this.s3Client.deleteObject(deleteParams).promise();
+        } catch (err) {
+            throw createAskSdkError(
+                this.constructor.name,
+                `Could not delete item (${objectId}) from bucket (${deleteParams.Bucket}): ${err.message}`,
```
Comment (pbheemag) [inline ask-sdk-s3-persistence-adapter/lib/attributes/persistence/S3PersistenceAdapter.ts]: extra comma at the end?

File: ask-sdk-s3-persistence-adapter/lib/attributes/persistence/S3PersistenceAdapter.ts [hunk 3]
```diff
@@ -109,4 +109,22 @@ export class S3PersistenceAdapter implements PersistenceAdapter {
             );
         }
     }
+
+    public async deleteAttributes(requestEnvelope : RequestEnvelope) : Promise<void> {
+        const objectId = path.join(this.pathPrefix, this.objectKeyGenerator(requestEnvelope));
+
+        const deleteParams : S3.DeleteObjectRequest = {
+            Bucket : this.bucketName,
+            Key : objectId,
```
Comment (pbheemag) [inline ask-sdk-s3-persistence-adapter/lib/attributes/persistence/S3PersistenceAdapter.ts]: extra comma at the end?

**Review by breedloj — APPROVED** [review]

**Pull request merged**